### PR TITLE
1498 add component tests for flutter

### PR DESCRIPTION
**Pull request by seluianova**
### Short description
- introduce flutter component testing
- add tests for favorites page
- add mocktail to dev dependencies for mocks creation

I have also included my attempt to fix an exception from #1312, but it didn't solve the issue. however, the 'await' was really missing, so I commited that change anyway

### Side effects
No

### Resolved issues
Fixes: #1498

### CODE REVIEW

File: frontend/lib/search/results_loader.dart [hunk 1]
```diff
@@ -102,7 +102,7 @@ class ResultsLoaderState extends State<ResultsLoader> {
         // Params are outdated.
         // If we're still at the first key, we must manually retrigger fetching.
         if (pageKey == _pagingController.firstPageKey) {
-          return _fetchPage(pageKey);
+          return await _fetchPage(pageKey);
         }
       }
       _pagingController.error = error;
```
Comment (michael-markl) [inline frontend/lib/search/results_loader.dart:108]: To fix the mentioned issue, we can probably do an early return when `!this.mounted` is true after every `await`. (But should probably be done in a separate PR :D )

Comment (seluianova) [reply frontend/lib/search/results_loader.dart:108]: Yea, I saw this approach but haven't tested it yet

File: frontend/test/favorites/favorites_page_test.dart [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                client: ValueNotifier(mockClient),
+                child: const FavoritesPage(),
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+
+    testWidgets('Favorite store is available', (WidgetTester tester) async {
```
Comment (f1sh1918) [inline frontend/test/favorites/favorites_page_test.dart:51]:
```suggestion
    testWidgets('shows store in favorites if the store is available', (WidgetTester tester) async {
```
i prefer to use a more descriptive descriptions to get an impression what will be tested

File: frontend/test/favorites/favorites_page_test.dart [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+      expect(find.text(t.favorites.title), findsOneWidget);
+
+      final acceptingStoreSummary = find.byType(AcceptingStoreSummary);
+      expect(acceptingStoreSummary, findsOneWidget);
+      expect(find.descendant(of: acceptingStoreSummary, matching: find.text('Test store')), findsOneWidget);
+      expect(find.descendant(of: acceptingStoreSummary, matching: find.text('Test description')), findsOneWidget);
+    });
+
+    testWidgets('Favorite store is no longer available', (WidgetTester tester) async {
```
Comment (f1sh1918) [inline frontend/test/favorites/favorites_page_test.dart:91]:
```suggestion
    testWidgets('shows a hint to the user if the store is not available anymore and offers the ability to remove it', (WidgetTester tester) async {
```

**Review by f1sh1918 — APPROVED** [review]
Super nice pr. Testing suite works great.
Just some minor suggestions for test description to make clear what will be tested. (optional)
